client/src/components/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,139 @@
+/* eslint-disable react/prop-types */
+
+import * as React from 'react';
+
+import Icon from './common/Icon';
+
+export interface Strings {
+  DASHBOARD: string;
+  EDIT_YOUR_ACCOUNT: string,
+  SEARCH: string,
+}
+
+export interface ModuleRenderContext {
+  slim: boolean;
+  expandingOrCollapsing: boolean;
+  currentPath: string;
+  strings: Strings;
+  navigate(url: string): Promise<void>;
+}
+
+export interface ModuleDefinition {
+  render(context: ModuleRenderContext): React.ReactFragment;
+}
+
+export interface SidebarProps {
+  modules: ModuleDefinition[];
+  currentPath: string;
+  strings: Strings;
+  navigate(url: string): Promise<void>;
+  onExpandCollapse?(collapsed: boolean);
+}
+
+export const Sidebar: React.FunctionComponent<SidebarProps> = (
+  { modules, currentPath, strings, navigate, onExpandCollapse }) => {
+  // 'collapsed' is a persistent state that is controlled by the arrow icon at the top
+  // It records the user's general preference for a collapsed/uncollapsed menu
+  // This is just a hint though, and we may still collapse the menu if the screen is too small
+  // Also, we may display the full menu temporarily in collapsed mode (see 'peeking' below)
+  const [collapsed, setCollapsed] = React.useState(window.innerWidth < 800);
```

Comment:
**Question :** do we want to collapse the menu when the user resizes the window and it gets past the 800 breakpoint ?

Reply:
Yep, I think we'll do that in a future PR though